pandas/core/groupby/groupby.py

```diff
@@ -1025,6 +1026,30 @@ def _apply_filter(self, indices, dropna):
             filtered = self._selected_obj.where(mask)  # Fill with NaNs.
         return filtered
 
+    @staticmethod
+    def _result_dtype(dtype, how) -> DtypeObj:
+        """
+        Get the desired dtype of a groupby result based on the
+        input dtype and how the aggregation is done.
+
+        Parameters
+        ----------
+        dtype : dtype, type
+            The input dtype of the groupby.
+        how : str
+            How the aggregation is performed.
+
+        Returns
+        -------
+        The desired dtype of the aggregation result.
+        """
+        d = {
+            (np.dtype(np.bool), "add"): np.dtype(np.int64),
```

Comment:
can you confirm that we are ok for min/max as well

Reply:
Yes - in that min/max on a boolean series remains bool. That is the same behavior as master.